### PR TITLE
Refactor Rules Protos

### PR DESCRIPTION
**Pull request by jtlisi**
**What this PR does**:

The `ruler/rules/rules.proto` file defines a proto for storing Prometheus rule groups. In order to make this proto message more amenable to the storage of rules, information that communicates the state of a running rule was removed and placed into a separate `GroupStateDesc` message in the `ruler/ruler.proto` file.

This PR executes the above change and updates all the consumers that utilize this change.

When rolling out this change, the experimental ruler API will be unable to serve the status of running rules until the rollout is complete. This will only affect the `/api/v1/rules` and `/api/v1/alerts` endpoint. It will not affect the evaluation of running rule groups.

**Checklist**
- [x] Tests updated
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

Signed-off-by: Jacob Lisi <jacob.t.lisi@gmail.com>

### CODE REVIEW

File: pkg/ruler/rules/rules.proto [hunk 1]
```diff
@@ -7,27 +7,25 @@ option go_package = "rules";
...
 import "google/protobuf/duration.proto";
-import "google/protobuf/timestamp.proto";
 import "github.com/cortexproject/cortex/pkg/ingester/client/cortex.proto";
 
 option (gogoproto.marshaler_all) = true;
 option (gogoproto.unmarshaler_all) = true;
 
 // RuleGroupDesc is a proto representation of a cortex rule group
 message RuleGroupDesc {
+  reserved 5;
```
Comment (pstibrany) [inline pkg/ruler/rules/rules.proto]: 5, 7, 8?

File: pkg/ruler/rules/rules.proto [hunk 2]
```diff
@@ -7,27 +7,25 @@ option go_package = "rules";
...
   repeated RuleDesc rules = 4;
-  bool deleted = 5;
   string user = 6;
-  google.protobuf.Timestamp evaluationTimestamp = 7  [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
-  google.protobuf.Duration evaluationDuration = 8 [(gogoproto.nullable) = false,(gogoproto.stdduration) = true];
 }
 
 // RuleDesc is a proto representation of a Prometheus Rule
 message RuleDesc {
+  reserved 7 to 10;
```
Comment (pstibrany) [inline pkg/ruler/rules/rules.proto]: to 12?

File: pkg/ruler/ruler.go [hunk 3]
```diff
@@ -551,14 +557,14 @@ func (r *Ruler) getLocalRules(userID string) ([]*rules.RuleGroupDesc, error) {
...
-			groupDesc.Rules = append(groupDesc.Rules, ruleDesc)
+			groupDesc.ActiveRules = append(groupDesc.ActiveRules, ruleDesc)
 		}
 		groupDescs = append(groupDescs, groupDesc)
 	}
 	return groupDescs, nil
 }
 
-func (r *Ruler) getShardedRules(ctx context.Context, userID string) ([]*rules.RuleGroupDesc, error) {
+func (r *Ruler) getShardedRules(ctx context.Context, userID string) ([]*GroupStateDesc, error) {
```
Comment (pstibrany) [inline pkg/ruler/ruler.go]: userID parameter is unused here

Comment (jtlisi) [reply pkg/ruler/ruler.go]: Makes sense, it should be safe to remove.

File: pkg/ruler/rules/rules.proto (hunk 1 above)
Comment (jtlisi) [reply pkg/ruler/rules/rules.proto]: 👍

File: pkg/ruler/rules/rules.proto (hunk 2 above)
Comment (jtlisi) [reply pkg/ruler/rules/rules.proto]: 👍

**Pull request merged**